src/include/rados.h

```diff
@@ -157,14 +157,16 @@ extern const char *ceph_osd_state_name(int s);
 #define CEPH_OSDMAP_REQUIRE_LUMINOUS (1<<18) /* require l for booting osds */
 #define CEPH_OSDMAP_RECOVERY_DELETES (1<<19) /* deletes performed during recovery instead of peering */
 #define CEPH_OSDMAP_PURGED_SNAPDIRS  (1<<20) /* osds have converted snapsets */
+#define CEPH_OSDMAP_PGLOG_HARDLIMIT  (1<<22) /* put a hard limit on pg log length */
```

Comment:
@neha-ojha @jdurgin @Yan-waller 

Yan and I just realized that we should also add this flag into the initial osdmap for any newly created clusters, right?  Otherwise new post-12.2.11 ceph clusters will turn this useful feature off by default.

Reply:
@xiexingguo @Yan-waller it's added to new clusters in nautilus and later. IIRC the reason this wasn't included in the luminous backport was for compatibility with existing mimic releases.

That is, there were mimic releases that did not support this, so if someone upgraded from luminous to those releases, the hardlimit would not be effective.

Reply:
yup, that's correct.

Reply:
@jdurgin @neha-ojha  
What if we enable this feature manually (e.g., because we finished upgrading from 12.2.9 to 12.2.11) and then upgrage to some mimic releases that claim to be not supporting this feature?

Reply:
@xiexingguo Here are the release notes I wrote for this scenario https://github.com/ceph/ceph/blob/master/doc/releases/mimic.rst#v1325-mimic, basically the safest path is to upgrade to 13.2.5

Reply:
@neha-ojha Aah, perfect! Thanks!